qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -385,6 +386,14 @@ def _mapper(self, circuit_graph, coupling_graph, trials=20):
                 self.property_set["final_layout"], circuit_graph.qubits
             )
 
+        layout_permutation = _inverse_pattern(layout.to_permutation(circuit_graph.qubits))
+
```

Comment:
Here and in other similar code things could be done more directly. I.e. instead of computing the final layout and converting it to a permutation, we could directly compute the permutation.